File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_06/java/Solution.java
package com.hi.dhl.algorithms.offer._06.java;

import com.hi.dhl.algorithms.model.ListNode;

import java.util.LinkedList;

/**
 * <pre>
 *     author: dhl
 *     desc  :
 * </pre>
 */
class Solution {

    public int[] reversePrint(ListNode head) {
        if (head == null) {
            return new int[]{};
        }
        int count = 0;
        ListNode currentNode = head;
        while (currentNode != null) {
            count++;
            currentNode = currentNode.next;
        }
        int[] reverse = new int[count];
        currentNode = head;
        while (currentNode != null) {
            reverse[--count] = currentNode.val;
            currentNode = currentNode.next;
        }
        return reverse;
    }
}

File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_06/kotlin/Solution.kt
package com.hi.dhl.algorithms.offer._06.kotlin

import com.hi.dhl.algorithms.model.ListNode


/**
 * <pre>
 *     author: dhl
 *     desc  :
 * </pre>
 */

class Solution {

    fun reversePrint(head: ListNode?): IntArray = head?.let {
        var count = 0
        var currentNode = head
        while (currentNode != null) {
            count++
            currentNode = currentNode.next
        }

        val reverse = IntArray(count) { 0 }
        currentNode = head
        while (currentNode != null) {
            reverse[--count] = currentNode.`val`
            currentNode = currentNode.next
        }
        reverse
    } ?: intArrayOf()
}